Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Nailin Guan, Yuyang Zhao
+-/
+import Mathlib.Algebra.Category.Grp.FiniteGrp
+import Mathlib.Topology.Algebra.ContinuousMonoidHom
+import Mathlib.Topology.Category.Profinite.Basic
+
+
+/-!
+
+# Category of Profinite Groups
+
+We say `G` is a profinite group if it is a topological group which is compact and totally
+disconnected.
+
+## Main definitions and results
+
```

Comment:
I would also mention in this introduction that you prove that a finite group can be seen as a profinite one (pointing to the correct declaration).